src/daft-local-execution/src/intermediate_ops/window_order_by_only.rs

```diff
@@ -0,0 +1,191 @@
+use std::{any::Any, sync::Arc};
+
+use common_error::{DaftError, DaftResult};
+use daft_core::prelude::*;
+use daft_dsl::{ExprRef, WindowExpr};
+use daft_micropartition::MicroPartition;
+use daft_recordbatch::RecordBatch;
+use itertools::Itertools;
+use tracing::{instrument, Span};
+
+use super::intermediate_op::{
+    IntermediateOpExecuteResult, IntermediateOpState, IntermediateOperator,
+    IntermediateOperatorResult,
+};
+use crate::ExecutionTaskSpawner;
+
+#[derive(Clone)]
+pub struct WindowOrderByOnlyParams {
+    window_exprs: Vec<WindowExpr>,
+    aliases: Vec<String>,
+    original_schema: SchemaRef,
+}
+
+pub struct WindowOrderByOnlyState {
+    record_batches: Vec<RecordBatch>,
+    params: WindowOrderByOnlyParams,
+    row_offset: u64,
+}
+
+impl WindowOrderByOnlyState {
+    fn new(params: WindowOrderByOnlyParams) -> Self {
+        Self {
+            record_batches: Vec::new(),
+            params,
+            row_offset: 0,
+        }
+    }
+
+    fn push(&mut self, input: Arc<MicroPartition>) -> DaftResult<()> {
+        let tables = input.get_tables()?;
+        for table in tables.iter() {
+            self.record_batches.push(table.clone());
+        }
+        Ok(())
+    }
+
+    fn drain_record_batches(&mut self) -> Vec<RecordBatch> {
+        std::mem::take(&mut self.record_batches)
+    }
+}
+
+impl IntermediateOpState for WindowOrderByOnlyState {
+    fn as_any_mut(&mut self) -> &mut dyn Any {
+        self
+    }
+}
+
+pub struct WindowOrderByOnlyOperator {
+    order_by: Vec<ExprRef>,
+    _descending: Vec<bool>,
+    params: WindowOrderByOnlyParams,
```

Comment:
```suggestion
    params: Arc<WindowOrderByOnlyParams>,
```

This way we only keep 1 copy of the params, but when we clone just increment the reference count